src/main/java/javaslang/collection/Iterator.java

```diff
@@ -323,7 +324,36 @@ public T next() {
 
     @Override
     default Iterator<T> dropWhile(Predicate<? super T> predicate) {
-        return null;
+        if (!hasNext()) {
```

Comment:
let's do `Objects.requireNonNull(predicate, "predicate is null")`. I've also forgotten in in various places and will add them later...